Все задания

RemigayloYegor opened the pull request.

RemigayloYegor committed:
Create README.md

RemigayloYegor committed:
Задания 1 и 2

RemigayloYegor committed:
Задание 3-1

RemigayloYegor committed:
Update 3-1.c

MarinaPaley commented on 1-2.c:
```diff
@@ -1,6 +1,7 @@
 #include<stdio.h>
 #include<math.h>
 #include<stdlib.h>
+
```
Для документрирования кода необходимо использовать [Doxygen](https://www.doxygen.nl/manual/docblocks.html).

MarinaPaley commented on 1-2.c:
```diff
@@ -1,6 +1,7 @@
+
 //Угол a задан в радианах. Найти его величину в градусах, минутах и секундах.
 
 double fun_scan();
```
Вам буковок не отсыпали? Пишите, пожалуйста, имена полностью.

![image](https://user-images.githubusercontent.com/26978698/143503209-23fadee5-c3a0-4cb2-9eb8-dc2e1bdffa79.png)

При ошибке сразу ее обрабатываем (пишем, что все плохо) и `abort();`

MarinaPaley commented on 1-3.c:
```diff
@@ -27,4 +27,4 @@ float fun_scan(){
     return 0;
-}
\ No newline at end of file
+}
```
`int U = 220` => `const double U = 220.0;`

`abort();`

Сначала проверка `I2`. затем расчет.

MarinaPaley commented on 2-2.c:
```diff
@@ -14,6 +14,7 @@ int main()
+
 float func(float x){
     float a = 1.1f;
     if (x == 1.2f){
```
Нельзя сравнивать значения переменных типа вещественное число (`float`, `double`)  на строгое равенство. За подробностями рекомендую обратиться к стандарту [`IEEE 754`](https://ru.wikipedia.org/wiki/IEEE_754-2008) или как минимум докладу [Андрея Акиньшина](https://aakinshin.net/ru/about/) ([видео на youtube (с нужным time code)](https://youtu.be/XEH0Hv54Fn8), [слайды (Ваш 18)](https://assets.ctfassets.net/9n3x4rtjlya6/6osesFo8pOyS8aUoOGUOWO/f0152328fc22be94f72231491ae586e7/Andrey_Akinshin_-_Lets_continue_to_talk_about_arithmetics.pdf), [слайды на SlideShare](https://www.slideshare.net/AndreyAkinshin/ss-70043876)).

https://replit.com/@MarinaPaley/HumongousKnottyFlashmemory

MarinaPaley commented on 3-1.c:
```diff
@@ -0,0 +1,40 @@
+float fun_prim(float x_min){
+  float prim;
+  prim = 3 * sin(sqrtf(x_min)) + 0.39 * x_min - 3.8;
+  if(prim != prim){
```
Проверьте (в отдельной функции) существование функции.
`return 3 * sin(sqrtf(x_min)) + 0.39 * x_min - 3.8;`

MarinaPaley commented on 3-1.c:
```diff
@@ -0,0 +1,40 @@
+  return 0;
+}
+
+float fun_prim(float x_min){
```
`float name(float x)`

RemigayloYegor committed:
Update 1-1.c

Добавил документацию и функцию

RemigayloYegor committed:
Update 1-2.c

Добавил документацию, изменил переменные, ABORT() и фунции

RemigayloYegor committed:
Update 2-3.c

добавил документацию, abort, переменные поменял

RemigayloYegor committed:
Update 3-1.c

Изменил проверку для существования Х. Добавил документацию

RemigayloYegor committed:
Update 1-3.c

Добавил документацию и убрал проверку в отдельную функцию. Переменные не менял, потому что это физические формулы

RemigayloYegor committed:
Update 2-2.c

Изменил проверку для double, откорректировал проверку вводимого числа и добавил документацию

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
+ * @param x,y,z значения параметров
+ * @return значение вычеслений
+ */
+double fun_calculator1(double x, double y, double z);
```
Необходимо использовать значащее название функции, например, FunctionA

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
+ * @param x,y,z значения параметров
+ * @return значение вычеслений
+ */
+double fun_calculator2(double x, double y, double z);
```
Название

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
-printf("x = %4.2f\n y = %4.2f\n z = %4.2f\n", x, y, z);
-printf("a = %f\n b = %f\n", a, b);
+  double x, y, z, a, b;
+  x = 0.29;
```
это константа
`const double x = 0.29;`

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
-printf("a = %f\n b = %f\n", a, b);
+  double x, y, z, a, b;
+  x = 0.29;
+  y = 3.7;
+  z = -1;
```
константы

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
+  x = 0.29;
+  y = 3.7;
+  z = -1;
+  a = fun_calculator1(x, y, z);
+  b = fun_calculator2(x, y, z);
+  printf("x = %4.2f\n y = %4.2f\n z = %4.2f\n", x, y, z);
+  printf("a = %f\n b = %f\n", a, b);
```
переменные a и b не нужны 
printf("a = %f\n b = %f\n", FunctionA(x, y, z), FunctionB(x, y, z));

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
+}
+
+double fun_calculator1(double x, double y, double z) {
+  double a;
+  a = 3 * pow(x, y) * log10(y) + exp(z * x);
+  return a;
```
return 3 * pow(x, y) * log10(y) + exp(z * x);

MarinaPaley commented on 1-1.c:
```diff
@@ -1,15 +1,38 @@
+}
+
+double fun_calculator2(double x, double y, double z) {
+  double b;
+  b = fabs(2 * y * z) / pow(sin(x), 2) - pow(x, 2) / 3;
+  return b;
```
return fabs(2 * y * z) / pow(sin(x), 2) - pow(x, 2) / 3;

RemigayloYegor committed:
1-1

RemigayloYegor committed:
3-2

RemigayloYegor committed:
4-1

RemigayloYegor committed:
Update 1-1.c

RemigayloYegor committed:
Create 4-2.c

MarinaPaley commented on 1-2.c:
```diff
@@ -1,31 +1,69 @@
+  a = scanf("%lf", &b);
+  if (a != 1) {
+    abort();
+  } else {
```
`else` не нужен

MarinaPaley commented on 1-2.c:
```diff
@@ -1,31 +1,69 @@
+  } else {
+    return b;
+  }
+  return 0;
```
Это зачем? Что-то не так

MarinaPaley commented on 1-2.c:
```diff
@@ -1,31 +1,69 @@
+}
+
+double fun_degrees(double corner) {
+  double degrees;
+  degrees = corner * 180 / M_PI;
```
1 строка

MarinaPaley reviewed (changes requested):
1. `convert` или `get`
2. Магические числа - не надо
